core/io/udp_server.cpp

```diff
@@ -89,6 +90,7 @@ Error UDPServer::poll() {
 Error UDPServer::listen(uint16_t p_port, const IP_Address &p_bind_address) {
 	ERR_FAIL_COND_V(!_sock.is_valid(), ERR_UNAVAILABLE);
 	ERR_FAIL_COND_V(_sock->is_open(), ERR_ALREADY_IN_USE);
+	ERR_FAIL_COND_V(p_port < 0 || p_port > 65535, ERR_INVALID_PARAMETER);
```

Comment:
```suggestion
	ERR_FAIL_COND_V_MSG(p_port < 1 || p_port > 65535, ERR_INVALID_PARAMETER, "The server port number must be between 1 and 65535 (inclusive).");
```

Reply:
`0` is a valid port for `listen/bind`. I'll add the `_MSG` though.